### DISCUSSION TITLE
Normalize trait

### DISCUSSION THREAD

**Discussion by woylie** [ory/kratos#2464] [Q&A] [answered]
Is there any way to normalize a trait value before saving it into the database?

I have a `phone_number` trait and would like to ensure that the value is stored as e164, without requiring the user to actually enter the value in that format. I set the `format` to `tel`, which ensures that the value is a valid phone number, but that of course does not format the value.

I'd like this to work in a settings flow for browsers (server-side application). Since the form action is a Kratos URL, I cannot do anything on the server side of the application to convert the value. I could add a bit of JavaScript and add a `submit` event handler to the form, but that would be less than ideal, since it runs on the client side.

**Comment by aeneasr** [comment, marked as answer]
Currently we only support validation. Normalization could probably be done using a webhook, once https://github.com/ory/kratos/pull/1585 is merged

**Reply by woylie** [reply]
That looks like it might work, thanks.
